inc/functions/files.php

```diff
@@ -1470,3 +1470,32 @@ function _rocket_is_windows_fs( $hard_reset = false ) { // phpcs:ignore WordPres
 function _rocket_get_wp_rocket_cache_path() { // phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedFunctionFound
 	return _rocket_normalize_path( rocket_get_constant( 'WP_ROCKET_CACHE_PATH' ) );
 }
+
+/**
+ * Get an array of php files in directory.
+ *
+ * @param string $dir_path Full path to the directory.
+ *
+ * @return array php files in the directory.
+ *               [...SplFileInfo file]
+ *
+ * @since  ver 3.6.1
+ *
+ * @author Caspar Green
+ */
+function _rocket_get_directory_php_files_array( $dir_path ) { // phpcs:ignore WordPress.NamingConventions.PrefixAllGlobals.NonPrefixedFunctionFound
+	try {
+		$config_dir = new FilesystemIterator( (string) $dir_path );
```

Comment:
A couple of things:

- Pass `FilesystemIterator::SKIP_DOTS` to the iterator.
- If we are making this function generic, then what gets returns is not always the `config_dir`. What will get returned? An iterator gets returned. Maybe here we want to standardize on `$iterator`. What do you think?

Reply:
The Docs say `DirectoryIterator` counts the dots, but `FilesystemIterator` extends `DirectoryIterator` and ignores them. Just in case, I included the dotfiles in the testdata array in the unit tests to make sure they're getting handled.

Reply:
Yes. This moving into an Iterator would be the next step.